src/Microsoft.Health.Fhir.Shared.Api/Controllers/FhirController.cs

```diff
@@ -512,5 +513,20 @@ public async Task<IActionResult> Versions()
 
             return new OperationVersionsResult(response, HttpStatusCode.OK);
         }
+
+        /// <summary>
+        /// Handles batch and transaction requests
+        /// </summary>
+        /// <param name="bundle">The bundle being posted</param>
+        [HttpPost]
+        [Route("", Name = RouteNames.PostBundle)]
+        [AuditEventType(AuditEventSubType.Batch)] // TODO: Our current auditing implementation only allows one audit event type attribute even though this action handles two.
```

Comment:
For transaction/batch, we will have to figure out the type based on the "type" property?

Reply:
I wonder if we can use routing constraints or something to separate these two out?

Reply:
Correct, the audit event type will be determined by the `type` of the bundle resource posted. 

I don't believe route constraints will work (at least out of the box) because [`IRouteConstraint` ](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.routing.irouteconstraint?view=aspnetcore-2.2) only contains the route data - not the payload.

There are at least a couple of options, but I think the decision merits more thought as a separate story:
- Move the auditing down to the mediator level
- Make the auditing attribute aware of payload
- Make the auditing a custom middleware

Reply:
We may be able to use an [`IActionConstraint`](https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.actionconstraints.iactionconstraint?view=aspnetcore-2.2) to achieve the route selection.